CONTRIBUTING.md

```diff
@@ -0,0 +1,50 @@
+### Setup
+
+We recommend that you use [development containers](https://containers.dev/) to contribute to this integration.
+
+Please refer tp the files in the `.devcontainer` folder for the specification of the development container for this repository.
+
+#### Installation
+
+To install all dependencies, run the following command:
+
+```shell
+poetry install --with test,dev,lint,test_integration
+```
+
+Note: This command is run automatically when you start the development container.
+
+### Testing
+
+To run the tests, use the following command:
+
+```shell
+make test
+```
+
+### Linting
+
+To run the linter, use the following command:
+
+```shell
+make format
+```
+
+### Creating a Pull Request
+
+1. Fork the repository.
+2. Create a new branch from the `main` branch.
+3. Make your changes.
+4. Run the tests and linter. Make sure the tests pass.
+5. Commit your changes.
+6. Push your branch to your fork.
+7. Create a pull request.
+8. In the pull request, describe your changes and tag the issue e.g. (#123) that it fixes.
```

Comment:
Maybe:

**if applicable,** tag the issue e.g. (#123) that it fixes.

Reply:
Going forward, there should be a discussion first before someone makes a PR. The discussion should happen in an Issue.

Reply:
This is to avoid the community doing duplicate work or making a contribution that is out of scope

Reply:
What do think of starting the "Creating a Pull Request" section with:

For minor changes, such as fixing typos in documentation or updating workflows and dependencies, feel free to directly submit a pull request (PR). For more substantial contributions like bug fixes or new features, we encourage you to first have a discussion with the project maintainers. This helps us align on the design of the solution and ensures your efforts are on the right track. You can do this by raising a new issue or volunteering to work on an existing one. Please wait to be assigned an issue before starting your work.

Reply:
Sounds like a good idea!